spec/controllers/concerns/omniauth_redirect_spec.rb

```diff
@@ -0,0 +1,36 @@
+require "rails_helper"
+
+# rubocop:disable Style/OpenStructUse
+# rubocop:disable Performance/OpenStruct
+RSpec.describe "Omniauth redirect", type: :request do
+  let!(:user) { create(:user) }
+  let!(:controller) { ApplicationController.new }
+  let!(:mock_warden) { Warden::Proxy.new({}, Warden::Manager.new(nil)) }
+
+  before do
+    allow(controller).to receive(:current_user).and_return(user)
+    allow(controller).to receive(:stored_location_for).and_return(nil)
+    allow(controller).to receive(:root_path).and_return("/new")
+  end
+
+  it "avoids i=i param in after_sign_in_path_for" do
+    mock_request = OpenStruct.new(env: { "warden" => mock_warden })
+    allow(controller).to receive(:request).and_return(mock_request)
+
+    path = controller.after_sign_in_path_for(user)
+    expect(path).not_to include("i=i")
+    expect(path).to end_with("/new?signin=true")
```

Comment:
This spec was built to test out `after_sign_in_path_for` response. This is a very tricky thing to achieve because all our omniauth responses are mocked in a way that we can't use many common patterns we use in our existing request/system specs.

If any of this raises a concern on anyone we can also just apply the patch without the spec, or I'll be happy to talk through in more detail the intricacies of testing this 🙂